solutions/cf-782b

```diff
@@ -0,0 +1,102 @@
+---
+id: cf-782B
+source: CF
+title: The Meeting Place Cannot Be Changed
+author: Aditya Gupta
+---
+
+[Official Analysis](https://codeforces.com/blog/entry/50854)
+
+## Explanation
+
+We can binary search on the location where the friends will meet up. Since we want a precision of $10^{-6}$, we stop when the difference between high and low is less than $10^{-6}$. 
+
+Whenever we are calculating the answer, we take note of which side the maximum answer came from. If the max answer comes from both sides, regardless of where we, decide to go next, our answer will only increase, so we return that answer. Else, if it comes from only one side of the point that we just calculated, we move our search towards that side, as moving it towards the other side will only increase our answer. 
+
+**Time Complexity:** $\mathcal{O}(N\log M)$ where $M$ is the farthest north location
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+typedef long long ll;
+
+vector<double> locations;
+vector<double> speeds;
+
+double min_ans = (double) INT_MAX, curr_ans;
+bool left_has_max = false;  // which side gives us max answer
+bool right_has_max = false;
+
+double solve(double loc) {
+	double curr_ans = 0;
```

Comment:
wtf you already declared ```curr_ans``` as double up there

also rename ```curr_ans``` to something better, like ```max_time```